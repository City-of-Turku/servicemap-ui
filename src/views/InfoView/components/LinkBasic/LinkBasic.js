import React from 'react';
import PropTypes from 'prop-types';
import { Typography, Link } from '@material-ui/core';

const LinkBasic = ({
  classes, intl, linkUrl, translationId,
}) => (
  <div className={classes.linkContainer}>
    <Link target="_blank" href={linkUrl}>
      <Typography className={classes.link} variant="body2" aria-label={intl.formatMessage({ id: translationId })}>
        {intl.formatMessage({ id: translationId })}
      </Typography>
    </Link>
  </div>
);

LinkBasic.propTypes = {
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  linkUrl: PropTypes.string,
  translationId: PropTypes.string,
};

LinkBasic.defaultProps = {
  linkUrl: '',
  translationId: '',
};

export default LinkBasic;
