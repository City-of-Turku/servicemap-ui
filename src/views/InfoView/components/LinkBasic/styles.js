export default theme => ({
  linkContainer: {
    paddingLeft: theme.spacing(3),
    paddingRight: theme.spacing(2),
    paddingTop: theme.spacing(1),
    paddingBottom: theme.spacing(1),
  },
  link: {
    color: theme.palette.link.main,
    textDecoration: 'underline',
  },
});
