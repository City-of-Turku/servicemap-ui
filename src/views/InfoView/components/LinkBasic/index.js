import LinkBasic from './LinkBasic';

export default LinkBasic;
