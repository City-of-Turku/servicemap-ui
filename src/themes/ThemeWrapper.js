import { ThemeProvider } from '@mui/material/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { useSelector } from 'react-redux';
import themes from '.';
import config from '../../config';
import { selectThemeMode } from '../redux/selectors/user';

// If external theme (by Turku) is true, then can be used to select which theme to get
const externalTheme = config.themePKG;
const isExternalTheme = !externalTheme || externalTheme === 'undefined' ? null : externalTheme;

/**
 * Returns default theme based on env value
 * @returns theme
 */
const getTheme = () => {
  if (isExternalTheme) {
    return themes.SMThemeTku;
  }
  return themes.SMTheme;
};

// Component to handle theme changes
const ThemeHandler = ({ children }) => {
  const themeMode = useSelector(selectThemeMode);
  return (// Get correct theme setting from store
    <ThemeProvider theme={themeMode === 'dark' ? themes.SMThemeDark : getTheme()}>
      {children}
    </ThemeProvider>
  );
};
ThemeHandler.propTypes = {
  children: PropTypes.node,
};
ThemeHandler.defaultProps = {
  children: null,
};

export default ThemeHandler;
