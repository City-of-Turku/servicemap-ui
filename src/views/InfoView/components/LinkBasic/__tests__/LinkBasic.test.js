// Link.react.test.js
import React from 'react';
import { ThemeProvider } from '@mui/material';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import themes from '../../../../../themes';
import LinkBasic from '../index';
import finnishTranslations from '../../../../../i18n/fi';

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnishTranslations,
};

const mockProps = {
  linkUrl: 'https://www.turku.fi/feedback',
  translationId: 'info.view.feedback.link',
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <IntlProvider {...intlMock}>
    <ThemeProvider theme={themes.SMTheme}>
      {children}
    </ThemeProvider>
  </IntlProvider>
);

const renderWithProviders = component => render(component, { wrapper: Providers });

describe('<LinkBasic />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<LinkBasic {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<LinkBasic {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0]).toBeInTheDocument();
  });

  it('does contain aria-label attribute', () => {
    const { container } = renderWithProviders(<LinkBasic {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].getAttribute('aria-label')).toBeTruthy();
  });
});
