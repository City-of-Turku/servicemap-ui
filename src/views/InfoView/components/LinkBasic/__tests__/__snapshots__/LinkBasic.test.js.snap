// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkBasic /> should work 1`] = `
<div>
  <div
    class="LinkBasic-linkContainer-1"
  >
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="https://www.turku.fi/feedback"
      target="_blank"
    >
      <p
        aria-label="www.turku.fi/palaute (linkki avautuu uudelle välilehdelle)."
        class="MuiTypography-root LinkBasic-link-2 MuiTypography-body2"
      >
        www.turku.fi/palaute (linkki avautuu uudelle välilehdelle).
      </p>
    </a>
  </div>
</div>
`;
